pkg/resources/deployment.go

```diff
@@ -0,0 +1,74 @@
+package resources
+
+import (
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Deployment(namespace string) *appsv1beta1.Deployment {
+	return &appsv1beta1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ark",
+			Namespace: namespace,
+		},
+		Spec: appsv1beta1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"component": "ark",
+					},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyAlways,
+					ServiceAccountName: "ark",
+					Containers: []corev1.Container{
+						{
+							Name:            "ark",
+							Image:           "gcr.io/heptio-images/ark:latest",
```

Comment:
We should parameterize either this entire value or at least the tag.

Reply:
I think tag would be sufficient and allows us to ensure some naive degree of provenance for API users. Happy to parameterize both path and tag or just the whole image though, it's your API surface at the end of the day.

Reply:
I know when we've done testing in the past, it's been nice to use a different image. Before we make a final decision, let's let @skriss weigh in too.

Reply:
What do you all think of potentially using [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) here. I am a big fan of this pattern for variadic API configuration.

Perhaps something like:

```go
// whole image
install.Deployment("heptio-ark", install.WithImage("gcr.io/heptio-images/ark:latest"))

// just the tag
install.Deployment("heptio-ark", install.WithTag("1.0.0"))
```